model/resolver/LOCResource.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+class LOCResource implements RemoteResource
+{
+    private $uri;
+
+    public function __construct(string $uri) {
+        $this->uri = $uri;
+    }
+    
+    public function resolve(int $timeout) : ?EasyRdf\Resource {
+        $graph = new EasyRdf\Graph($this->uri);
+        // guess the concept scheme based on the URI
+        if (preg_match('|(http://id.loc.gov/[^/]+/[^/]+)/.*|', $this->uri, $matches)) {
+            $graph->addResource($this->uri, 'skos:inScheme', $matches[1]);
+        }
+
+        $opts = array('http' => array('method'=>'HEAD',
+                                      'user_agent' => 'Skosmos',
+                                      'timeout' => $timeout));
+        $context  = stream_context_create($opts);
+        $fd = fopen($this->uri, 'rb', false, $context);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $fd. Configured minimum length is 3.](https://app.codacy.com/manual/NatLibFi/Skosmos/pullRequest?prid=4951447)